dali/test/python/test_operator_get_property.py

```diff
@@ -0,0 +1,138 @@
+    p.build()
+    output = p.run()
+    for out in output:
+        out = out if device == 'cpu' else out.as_cpu()
+        for source_info, ref in zip(out, ref_paths):
+            assert _uint8_tensor_to_string(source_info) == ref
+
+
+def test_file_properties():
+    for dev in ['cpu', 'gpu']:
+        yield _test_file_properties, dev
+
+
+@pipeline_def
+def wds_properties(root_path, device):
+    read = fn.readers.webdataset(paths=[root_path], ext=['jpg'])
+    if device == 'gpu':
+        read = read.gpu()
+    return fn.get_property(read, key="source_info")
+
+
+def _test_wds_properties(device):
+    root_path = os.path.join(get_dali_extra_path(), "db/webdataset/MNIST/devel-0.tar")
+    ref_offset = [1536, 4096, 6144, 8704, 11264, 13824, 16384, 18432]
+    p = wds_properties(root_path, device, batch_size=8, num_threads=4, device_id=0)
+    p.build()
+    output = p.run()
+    for out in output:
+        out = out if device == 'cpu' else out.as_cpu()
+        for source_info, offset in zip(out, ref_offset):
+            assert _uint8_tensor_to_string(
+                source_info) == f"archive {root_path}tar file at \"{root_path}\"component offset {offset}"
+
+
+def test_wds_properties():
+    for dev in ['cpu', 'gpu']:
+        yield _test_wds_properties, dev
+
+
+@pipeline_def
+def tfr_properties(root_path, index_path, device):
+    import nvidia.dali.tfrecord as tfrec
+    inputs = fn.readers.tfrecord(path=root_path, index_path=index_path,
+                                 features={"image/encoded": tfrec.FixedLenFeature((), tfrec.string, ""),
+                                           "image/class/label": tfrec.FixedLenFeature([1], tfrec.int64, -1)})
+    if device == 'gpu':
+        return fn.get_property(inputs["image/encoded"], key="source_info").gpu(), \
+               fn.get_property(inputs["image/class/label"], key="source_info").gpu()
+    return fn.get_property(inputs["image/encoded"], key="source_info"), \
+           fn.get_property(inputs["image/class/label"], key="source_info")
+
+
+def _test_tfr_properties(device):
+    root_path = os.path.join(get_dali_extra_path(), 'db', 'tfrecord', 'train')
+    index_path = os.path.join(get_dali_extra_path(), 'db', 'tfrecord', 'train.idx')
+    idx = [0, 171504, 553687, 651500, 820966, 1142396, 1380096, 1532947]
+    p = tfr_properties(root_path, index_path, device, batch_size=8, num_threads=4, device_id=0)
+    p.build()
+    output = p.run()
+    for out in output:
+        out = out if device == 'cpu' else out.as_cpu()
+        for source_info, ref_idx in zip(out, idx):
+            assert _uint8_tensor_to_string(source_info) == f"{root_path} at index {ref_idx}"
+
+
+def test_tfr_properties():
+    for dev in ['cpu', 'gpu']:
+        yield _test_tfr_properties, dev
+
+
+@pipeline_def
+def es_properties(layouts, device):
+    num_outputs = len(layouts)
+
+    def gen_data():
+        from torch import rand
```

Comment:
this would make anyone running this test to have torch installed. Why not using `np.random` utils?

Reply:
In this case it should go to `test_pytorch.sh` while it doesn't need to.

Reply:
Done